conn2res/plotting.py

```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf-8 -*-
+"""
+Plotting functions
+
+@author: Estefany Suarez
+"""
+
+import os
+import seaborn as sns
+import matplotlib.pyplot as plt
+import numpy as np
+from numpy.linalg import svd
+
+PROJ_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+FIG_DIR = os.path.join(PROJ_DIR, 'figs')
+if not os.path.isdir(FIG_DIR):
+    os.makedirs(FIG_DIR)
+
+
+class PCA:
+    """
+    Class that represents a simplified PCA object
+
+    key features:
+    - specifying the index of principal components we want to keep
+    - principal components either unnormalized/normalized by singular values
+
+    TODO
+    """
+
+    def __init__(self, idx_pcs=None, n_pcs=None, **kwargs):
+        """
+        Constructor class for time series
+        """
+        # indexes of principal components to keep
+        if idx_pcs is not None:
+            if isinstance(idx_pcs, list):
+                idx_pcs = np.array(idx_pcs)
+            if isinstance(idx_pcs, int):
+                idx_pcs = np.array([idx_pcs])
+            self.idx_pcs = idx_pcs
+            self.n_pcs = len(self.idx_pcs)
+
+        # number of principal components to keep
+        if n_pcs is not None:
+            self.setdefault('n_pcs', n_pcs)
+
+    def setdefault(self, attribute, value):
+        # add attribute (with given value) if not existing
+        if not hasattr(self, attribute):
+            setattr(self, attribute, value)
+
+    def fit(self, data, full_matrices=False, **kwargs):
+        # fit PCA
+        self.u, self.s, self.vh = svd(data, full_matrices=full_matrices)
+
+        # set number of principal components if not existing
+        self.setdefault('n_pcs', self.s.size)
+
+        # set indexes of principal components if not existing
+        self.setdefault('idx_pcs', np.arange(self.n_pcs))
+
+        return self
+
+    def transform(self, data, normalize=False, **kwargs):
+        # transform data into principal components
+        pc = (data @ self.vh.T[:, self.idx_pcs]).reshape(-1, self.n_pcs)
+
+        # normalize principal components by singular values (loop for efficiency)
+        if normalize == True:
+            for i in range(self.n_pcs):
+                pc[:, i] /= self.s[self.idx_pcs[i]]
+
+        return pc
+
+    def fit_transform(self, data, **kwargs):
```

Comment:
I tried to keep the general python nomenclature with fit/transform/fit_transform methods.